src/drivers/intel/dmic/dmic_nhlt.c

```diff
@@ -272,21 +278,22 @@ int dmic_set_config_nhlt(struct dai *dai, void *spec_config)
 	/* Write the FIFO control registers. The clear/set of bits is the same for
 	 * all DMIC_HW_VERSION
 	 */
-	if (dai->index == 0) {
-		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
-		val = (out_control[0] &
-			~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
-			OUTCONTROL0_FINIT_BIT;
+	/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
+	val = (out_control[dai->index] &
+		~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
+		OUTCONTROL0_FINIT_BIT;
```

Comment:
this is because those `OUTCONTROL0` and `OUTCONTROL1` bits are the same? Why do we have two of them then? Wouldn't it be enough to have just one like `OUTCONTROL_TIE_BIT` etc.? Or at least add a comment here, that bit numbers are the same. But I see that we already use bit definitions for `OUTCONTROL0` for both FIFOs elsewhere in the driver

Reply:
Yep, the bits are the same in the registers and I don't see reason for them to become different in the future. This followed the HW specification but it's a duplicate definition. If you prefer I could add a commit that makes OUTCONTROL0 just OUTCONTROL and removes OUTCONTROL1.